.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+  merge_group:
+
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  NUGET_PACKAGES: ${{ github.workspace }}/.nuget/packages
+
+jobs:
+  prefetch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.1
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4.0.0
+        with:
+          dotnet-version: 8
+
+  examples:
+    needs: [prefetch]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        example:
+          - Audit/AuditLog
+          - Audit/AuditLogBulk
+          - Audit/AuditLogBulkAsync
+          - Audit/AuditLogCustomSchema
+          - Audit/AuditLogNsign
+          - Audit/AuditMulticonfig
+          - Audit/AuditRoot
+          - Audit/AuditSearch
+          - Audit/AuditSearchCustomSchema
+          - AuthN/InviteActions
+          - AuthN/UserActions
+          - Embargo/IPCheck
+          - Embargo/ISOCheck
+          - FileScan/Scan_Async
+          - FileScan/Scan_RequestUploadURL_Post
+          - FileScan/Scan_RequestUploadURL_Put
+          - FileScan/Scan_Sync
+          - Intel/Defang/Defang
+          - Intel/Domain/Reputation
+          - Intel/Domain/ReputationBulk
+          - Intel/File/Reputation
+          - Intel/File/ReputationBulk
+          - Intel/File/ReputationFilepath
+          - Intel/IP/Geolocate
+          - Intel/IP/GeolocateBulk
+          - Intel/IP/GetDomain
+          - Intel/IP/GetDomainBulk
+          - Intel/IP/IsProxy
+          - Intel/IP/IsProxyBulk
+          - Intel/IP/IsVPN
+          - Intel/IP/IsVPNBulk
+          - Intel/IP/Reputation
+          - Intel/IP/ReputationBulk
+          - Intel/URL/Reputation
+          - Intel/URL/ReputationBulk
+          - Intel/User/PasswordBreached
+          - Intel/User/UserBreachedByEmail
+          - Intel/User/UserBreachedByIP
+          - Intel/User/UserBreachedByPhone
+          - Intel/User/UserBreachedByUsername
+          - Redact/RedactMulticonfig
+          - Redact/Structured
+          - Redact/Text
+          - Vault/Encrypt
+          - Vault/EncryptStructured
+          - Vault/Rotate
+          - Vault/Sign
+        dotnet-version: [6, 7, 8]
+    defaults:
+      run:
+        working-directory: ./examples/${{ matrix.example }}
+    steps:
+      - name: Checkout code
```

Comment:
doesn't the prefetch do this?

Reply:
Each job starts fresh, so gotta checkout again. The prefetch is there is cache things like dependencies (which I couldn't really accomplish for this repo because we're not using lockfiles).